### PR TITLE
refactor: improve branch support logic

### PR DESCRIPTION
**Pull request by codebytere**
Secondary to https://github.com/electron/trop/pull/300

Cleans up some comment-initiated backport logic

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -546,56 +546,55 @@ const probotHandler: ApplicationFunction = async (robot, { getRouter }) => {
           const branches = targetBranches.split(',').map((b) => b.trim());
           for (const branch of branches) {
             robot.log(
-              `Initiating backport to \`${branch}\` from 'backport-to' comment`,
+              `Attempting backport to \`${branch}\` from 'backport-to' comment`,
             );
 
-            if (!branch.trim()) continue;
```
Comment (ckerr) [inline src/index.ts:552]: Agree that the `trim()` call is redundant after `b.trim()`, but do we need to still check whether `branch` is empty? Or add a `.filter()` after `.map()` to remove empty strings?

File: src/index.ts [hunk 2]
```diff
@@ -548,55 +548,55 @@ const probotHandler: ApplicationFunction = async (robot, { getRouter }) => {
...
+              );
+              await context.octokit.issues.createComment(
+                context.repo({
+                  body: `Provided branch \`${branch}\` does not appear to exist.`,
+                  issue_number: issue.number,
+                }),
+              );
+              continue;
+            }
+
```
Comment (jkleinsc) [inline src/index.ts:583]: Why is this block moved under the section flagged by NO_EOL_SUPPORT?

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Maybe I am missing something here, but it looks like this change moves the check for branch existence to only fire when NO_EOL_SUPPORT is set?

File: src/index.ts (hunk 2 above)
Comment (codebytere) [reply src/index.ts:583]: `getSupportedBranches` itself checks branch existence - so in moving it down we save a redundant check in the case the variable is set.

**Comment by codebytere** [comment]
@jkleinsc 

>  check for branch existence to only fire when NO_EOL_SUPPORT is set?

it moves the conditional down, but not inside the check - if `NO_EOL_SUPPORT` the check still runs.

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**